src/components/Taxonomy/Taxonomy.tsx

```diff
@@ -454,14 +451,47 @@ const Taxonomy = ({
   }, [externalSelected]);
 
   useEffect(() => {
-    if (isOpen) {
-      document.body.addEventListener("click", onClickOutside, true);
-      document.body.addEventListener("keydown", onEsc);
-    } else {
+    document.body.addEventListener("click", onClickOutside, true);
+    document.body.addEventListener("keydown", onKeyDown);
+    return () => {
       document.body.removeEventListener("click", onClickOutside);
-      document.body.removeEventListener("keydown", onEsc);
-    }
-  }, [isOpen]);
+      document.body.removeEventListener("keydown", onKeyDown);
```

Comment:
it's better to move declaration before use

Reply:
not sure I follow